app/_src/gateway/reference/wasm.md

```diff
@@ -0,0 +1,141 @@
+---
+title: WebAssembly
+content_type: reference
+badge: free
+---
+
+## WebAssembly in Kong
+
+In addition to Lua plugins, {{site.base_gateway}} supports WebAssembly as a
+means of extending proxy functionality and performing custom business logic.
+This is made possible by Kong's [open source implementation](https://github.com/Kong/ngx_wasm_module) of
+the [Proxy-Wasm specification](https://github.com/proxy-wasm/spec).
+
+## Concepts
+
+### Filter
+
+In the context of Proxy-Wasm, a "filter" refers to a specific component or
+module that is executed at some stage of request/response handling. Filters can
+perform actions such as:
+
+* read or write request or response headers
+* read or write the request or response body
+* terminate the request early and send a self-generated response
+
+If you are familiar with Kong's Lua plugin support, you can think of filters
+as semi-analogous to plugins.
+
+Filters differ from Lua plugins in their file format at runtime. A Lua plugin
+typically spans several different Lua source code files, all packaged and
+deployed in text format. Filters must be compiled from source code
+into a single binary file with the `.wasm` extension.
+
+In order to be usable in {{site.base_gateway}}, all filter files must be present
+on disk in a single directory, denoted by the `wasm_filters_path` configuration
+directive.
+
+### Filter chain
+
+A filter chain is the database entity representing one or more filters executed
+for each request to a particular service or route, each one with its
+configuration.
+
+Example:
+
+```yaml
+---
+filter-chains:
+  name: my-filter-chain
+  # filter chains can be toggled on/off for testing
+  enabled: true
+  filters:
+    # this maps to $wasm_filters_path/my_filter.wasm
+    - name: my_filter
+      # at runtime, configuration is passed to the filter as a byte array
+      #
+      # filter configuration is inherently untyped--schema and
+      # encoding/serialization are left up to the filter code itself
+      config: "foo=bar baz=bat"
+
+    - name: my_other_filter
+      # individual filters within a chain can be toggled on/off for testing
+      enabled: false
+      config: >-
+          {
+            "some_settings": {
+              "a": "b",
+              "c": "d"
+            },
+            "foo": "bar"
+          }
+
+    # the same filter may occur more than once in a single chain, and all   entries will be
+    # executed
+    - name: my_filter
+      config: "foo=123 baz=456"
+```
+
+
+#### Relationships
+
+Filter chains **must** be linked to a service or route, and as of this writing,
```

Comment:
```suggestion
Filter chains **must** be linked to a Service or Route, and as of this writing,
```

There are many other instances of model entities not capitalized in the diff; this is a just for a "catch-all".